### ISSUE TITLE
Compiler doesn't report warning when a struct field is not initialized that contains non-nullable reference types

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#47097]
**Version Used**: 

Version 16.6.0 Preview 2.0 [29915.30.master]

**Steps to Reproduce**:

```C#

#nullable enable

readonly struct S<T> where T : class
{
    public readonly T X;

    public S(T x)
    {
        X = x;
    }
}

class C<T> where T : class
{
    public S<T> S;

    public C()
    {
    }

    public void F(S<T> s)
    {
        S = s;
    }
}
```

**Expected Behavior**:

Warning - The type S<T> of field C.S has non-nullable members but is not initialized in the constructor of C.

**Actual Behavior**:

No warning.

**Comment by tmat** [comment]
@RikkiGibson

**Comment by RikkiGibson** [comment]
> Version 16.6.0 Preview 2.

Did you mean 16.8 preview 2?

**Comment by RikkiGibson** [comment]
Whether a struct contains nullable reference types is not considered in nullable analysis in either the "old" or "new" nullable constructor analysis. I do think we need to find a way to plug this gap, possibly with "non-defaultable" structs or a similar proposal, but that decision will have to come from the language design side.

**Comment by tmat** [comment]
> Did you mean 16.8 preview 2? 

Either one :)
